student-management-system/src/main/java/com/example/service/impl/StaffManagementServiceImpl.java

```diff
@@ -0,0 +1,131 @@
+        Staff staff = staffManagementDao.staffAccess(staffDto.getUsername());
+        return passwordEncoder.matches(staffDto.getPassword(), staff.getPassword());
+    }
+
+    @Override
+    public List<StaffDto> listStaff() throws ServiceException {
+        List<StaffDto> staffDtoList = new ArrayList<>();
+        try {
+            List<Staff> staffList = staffManagementDao.listStaff();
+            staffList.parallelStream().forEach(staff -> {
+
+                    StaffDto staffDto = new StaffDto();
+                    if(staff.getStatus().equals("ACTIVE")){
+                        System.out.println("inside if condition");
+                    staffDto.setImageURL(staff.getImageURL());
+                    staffDto.setUsername(staff.getUsername());
+                    staffDto.setPassword(staff.getPassword());
+                    staffDto.setId(staff.getId());
+                    staffDto.setName(staff.getName());
+                    staffDto.setDepartment(staff.getDepartment());
+                    staffDto.setStaffID(staff.getStaffID());
+                    staffDtoList.add(staffDto);
+                }
+                    else{
+                        System.out.println(staff.getStatus());
+                    }
+
+            });
+        } catch (MongoDbException mongoDbException) {
+            throw new ServiceException("Error while processing data", mongoDbException);
+        }
+        return staffDtoList;
+    }
+
+    @Override
+    public StaffDto findStaffById(String id) throws ServiceException {
+        StaffDto staffDto = new StaffDto();
+        try {
+            Staff staff = staffManagementDao.findStaffById(id);
+            if(staff.getStatus().equals("ACTIVE")) {
+                staffDto.setImageURL(staff.getImageURL());
+                staffDto.setId(staff.getId());
+                staffDto.setName(staff.getName());
+                staffDto.setDepartment(staff.getDepartment());
+                staffDto.setStaffID(staff.getStaffID());
+                staffDto.setUsername(staff.getUsername());
+                staffDto.setPassword(staff.getPassword());
+            }
+        } catch (NotFoundException notFoundException) {
+            throw new ServiceException("Staff not found", notFoundException);
+        }
+        return staffDto;
+    }
+
+    @Override
+    public void addStaff(StaffDto staffDto) throws ServiceException {
+        Staff staff = new Staff();
+
+            if((staffDto.getUsername().length() > 3 ) && (staffDto.getPassword().length() > 3) && (staffManagementDaoImpl.findStaffByUserName(staffDto.getUsername()) == null)){
+                String encodedPassword = passwordEncoder.encode(staffDto.getPassword());
+//                System.out.println(staffDto.getName().length());
+                staff.setName(staffDto.getName());
+                staff.setDepartment(staffDto.getDepartment());
+                staff.setStaffID("STF" + ((int)(Math.random()*(1000 - 100) + 100)));
```

Comment:
this is flow used in student also make this as a method and pass the what id you need and get from there